contracts/oracle/CollateralizationOracle.sol

```diff
@@ -267,32 +279,32 @@ contract CollateralizationOracle is ICollateralizationOracle, CoreRef {
             Decimal.D256 memory _oraclePrice = Decimal.zero();
 
             // For each deposit...
-            uint256 _nTokens = 0;
+            uint256 _totalTokenBalance  = 0;
             for (uint256 j = 0; j < tokenToDeposits[_token].length; j++) {
                 address _deposit = tokenToDeposits[_token][j];
 
-                // ignore deposits that are paused
-                if (!IPausable(_deposit).paused()) {
-                  // On the first unpaused deposit, read the oracle.
-                  // This is done inside the loop, after _deposit.paused() check,
-                  // because if all deposits of an asset are paused, there is no
-                  // need to read the oracle.
-                  if (!_oracleRead) {
-                    (_oraclePrice, _oracleValid) = IOracle(tokenToOracle[_token]).read();
-                    _oracleRead = true;
-                    if (!_oracleValid) {
-                      validityStatus = false;
+                // ignore deposits that are excluded by the Guardian
+                if (!excludedDeposits[_deposit]) {
+                    // On the first inspected deposit, read the oracle.
+                    // This is done inside the loop, after exclusion check,
+                    // because if all deposits of an asset are excluded, there is
+                    // no need to read the oracle.
+                    if (!_oracleRead) {
+                        (_oraclePrice, _oracleValid) = IOracle(tokenToOracle[_token]).read();
```

Comment:
Would make more sense to put the read call once outside the loop right?

Reply:
it used to be above the loop, but if all deposits of this token are excluded, there is no need to fetch the oracle, that's why I put it in the loop.

I'll do even better, and fetch the oracle after the loop, if the tokenBalance is != 0